server/src/main/java/org/elasticsearch/action/bulk/FailureStoreMetrics.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.bulk;
+
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.Map;
+
+/**
+ * A record APM metrics that concern failure stores.
+ */
+public record FailureStoreMetrics(LongCounter failureCounter, LongCounter redirectCounter) {
```

Comment:
I saw this approach in [RepositoriesMetrics.java](https://github.com/nielsbauman/elasticsearch/blob/08b687811baac42d7d105f0fa9aca98882728534/server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java#L15-L15) and I thought it looked clean. Would you prefer a regular class?

Reply:
It looks like RepositoriesMetrics is only ever used to hold the counters. Since we're expecting people to increment the counters in the FailureStoreMetrics via the methods on it, we probably shouldn't expose the counters via getters. I think a regular class makes more sense here if we make the incrementing methods non-static.